doc: Review and add tests on `ResolveTargetDocumentListener` (#2660)
doctrine/mongodb-odm: docs/en/cookbook/resolve-target-document-listener.rst

Keeping Your Modules Independent
================================

> If you work with independent modules, you may encounter the problem of creating

Reviewer: In case you want to add an example, Sylius uses this concept extensively to allow for customising entity classes.

Author: I don't know Sylius to know what to write. I haven't found any reference to `ResolveTargetDocumentListener` or `ResolveTargetEntityListener` in the code or in the doc of Sylius.

Configuration
-------------

We're going to use the following basic documents to explain how to set up and
use the ``ResolveTargetDocumentListener``.

A ``Customer`` class in the ``CustomerModule``. This class will be extended in
the application:

.. code-block:: php

    <?php

    namespace Acme\CustomerModule\Document;

    #[Document]
    class Customer
    {
        #[Id]
        public string $id;

        #[Field]
        public string $name;
    }

An ``Invoice`` document in the ``InvoiceModule``:

.. code-block:: php

    <?php

    namespace Acme\InvoiceModule\Document;

    use Acme\InvoiceModule\Model\InvoiceSubjectInterface;

    #[Document]
    class Invoice
    {
        #[Id]
        public string $id;

        #[ReferenceOne]
        public InvoiceSubjectInterface $subject;
    }

> This class has de reference to the ``InvoiceSubjectInterface``. This interface

Reviewer: ```suggestion
This class has a reference to an ``InvoiceSubjectInterface``. This interface
```

Author: https://github.com/doctrine/mongodb-odm/pull/2667
